numpy/distutils/ccompiler.py

```diff
@@ -191,7 +230,8 @@ def CCompiler_compile(self, sources, output_dir=None, macros=None,
 
     def single_compile(args):
         obj, (src, ext) = args
-        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+        if _needs_build(obj):
+            self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
```

Comment:
The issue vs. #9035 is probably that `_needs_build` should also check if any compilation flags differ from the previous compilation.